packages/uauth/src/index.ts

```diff
@@ -0,0 +1,336 @@
+      getInterface: async ({ chains, EventEmitter }) => {
+        const uauthInstance = new UAuth({
+          clientID: clientID,
+          redirectUri: redirectUri,
+          scope: scope
+        })
+
+        let user: UserInfo
+        try {
+          user = await uauthInstance.user()
+        } catch (error) {
+          if (!uauthInstance.fallbackLoginOptions.scope.includes('wallet')) {
+            throw new Error(
+              'Must request the "wallet" scope for connector to work.'
+            )
+          }
+
+          if (shouldLoginWithRedirect) {
+            await uauthInstance.login()
+            // NOTE: We don't want to throw because the page will take some time to
+            // load the redirect page.
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            await new Promise<void>(() => {})
+            // We need to throw here otherwise typescript won't know that user isn't null.
+            throw new Error('Should never get here.')
+          } else {
+            await uauthInstance.loginWithPopup()
+            user = await uauthInstance.user()
+          }
+        }
+
+        if (!user.wallet_type_hint) {
+          throw new Error('no wallet type hint')
+        }
+
+        // @ts-ignore
+        let provider: EIP1193Provider | EthProvider
+
+        if (['web3', 'injected'].includes(user.wallet_type_hint)) {
+          provider =
+            (window as any).ethereum ||
+            ((window as any).web3 && (window as any).web3.currentProvider)
+          provider = createEIP1193Provider(provider)
+        } else if (user.wallet_type_hint === 'walletconnect') {
+          const { StaticJsonRpcProvider } = await import(
+            '@ethersproject/providers'
+          )
+
+          const { ProviderRpcError, ProviderRpcErrorCode } = await import(
+            '@web3-onboard/common'
+          )
+
+          const { default: WalletConnect } = await import(
+            '@walletconnect/client'
+          )
+
+          // This is a cjs module and therefor depending on build tooling
+          // sometimes it will be nested in the { default } object and
+          // other times it will be the actual import
+          // @ts-ignore - It thinks it is missing properties since it expect it to be nested under default
+          let QRCodeModal: typeof import('@walletconnect/qrcode-modal').default =
+            await import('@walletconnect/qrcode-modal')
```

Comment:
```suggestion
          let QRCodeModal = (await import('@walletconnect/qrcode-modal')).default
```
Does something like this help TS resolve without the need for ts-ignore?

Reply:
ah gotcha! If this doesn't fix the ts error no worries 🤓

Reply:
Worked! Will add it to the commit with the option validation